pkg/standalone-utils/test/foundry/BalancerFeeBurner.t.sol

```diff
@@ -0,0 +1,259 @@
+        assertEq(daiVaultBalanceAfter, daiVaultBalanceBefore + TEST_BURN_AMOUNT, "DAI balance should increase (vault)");
+        assertEq(
+            usdcVaultBalanceAfter,
+            usdcVaultBalanceBefore - TEST_BURN_AMOUNT,
+            "USDC balance should decrease (vault)"
+        );
+        assertEq(dai.balanceOf(address(feeBurner)), 0, "FeeBurner should not hold DAI");
+        assertEq(usdc.balanceOf(address(feeBurner)), 0, "FeeBurner should not hold USDC");
+    }
+
+    function testBurnWithMultiHop() external {
+        IBalancerFeeBurner.SwapPathStep[] memory steps = new IBalancerFeeBurner.SwapPathStep[](2);
+        steps[0] = IBalancerFeeBurner.SwapPathStep({ pool: daiWethPool, tokenOut: weth });
+        steps[1] = IBalancerFeeBurner.SwapPathStep({ pool: wethUsdcPool, tokenOut: usdc });
+
+        uint256 daiBalanceBefore = dai.balanceOf(alice);
+        uint256 usdcBalanceBefore = usdc.balanceOf(alice);
+        uint256 daiVaultBalanceBefore = dai.balanceOf(address(vault));
+        uint256 usdcVaultBalanceBefore = usdc.balanceOf(address(vault));
+
+        vm.startPrank(alice);
+        feeBurner.setBurnPath(dai, steps);
+
+        IERC20(address(dai)).forceApprove(address(feeBurner), TEST_BURN_AMOUNT);
+        feeBurner.burn(address(0), dai, TEST_BURN_AMOUNT, usdc, MIN_TARGET_TOKEN_AMOUNT, alice, orderDeadline);
+        vm.stopPrank();
+
+        uint256 daiBalanceAfter = dai.balanceOf(alice);
+        uint256 usdcBalanceAfter = usdc.balanceOf(alice);
+        uint256 daiVaultBalanceAfter = dai.balanceOf(address(vault));
+        uint256 usdcVaultBalanceAfter = usdc.balanceOf(address(vault));
+
+        assertEq(daiBalanceAfter, daiBalanceBefore - TEST_BURN_AMOUNT, "DAI balance should decrease (alice)");
+        assertEq(usdcBalanceAfter, usdcBalanceBefore + TEST_BURN_AMOUNT, "USDC balance should increase (alice)");
+        assertEq(daiVaultBalanceAfter, daiVaultBalanceBefore + TEST_BURN_AMOUNT, "DAI balance should increase (vault)");
+        assertEq(
+            usdcVaultBalanceAfter,
+            usdcVaultBalanceBefore - TEST_BURN_AMOUNT,
+            "USDC balance should decrease (vault)"
+        );
+        assertEq(dai.balanceOf(address(feeBurner)), 0, "FeeBurner should not hold DAI");
+        assertEq(usdc.balanceOf(address(feeBurner)), 0, "FeeBurner should not hold USDC");
+    }
+
+    function testBurnRevertIfNotAuthorized() external {
+        vm.expectRevert(IAuthentication.SenderNotAllowed.selector);
+        feeBurner.burn(address(0), dai, TEST_BURN_AMOUNT, usdc, MIN_TARGET_TOKEN_AMOUNT, alice, orderDeadline);
+    }
+
+    function testSetPathRevertIfNotAuthorized() external {
+        vm.expectRevert(IAuthentication.SenderNotAllowed.selector);
+        feeBurner.setBurnPath(dai, new IBalancerFeeBurner.SwapPathStep[](0));
+    }
+
+    function testBurnRevertIfDeadlinePassed() external {
```

Comment:
Let's also have a test that reverts if we get less than the min specified amount for the target token